### PR TITLE
Use the kernel to check for ipykernel 6 version, instead of the python installer

### PR DESCRIPTION
**Pull request by DavidKutu**
This works for both local and remote.

For #7576

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/debugger/jupyter/debuggingManager.ts [hunk 1]
```diff
@@ -421,31 +411,35 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
...
+                    kernel = this.kernelProvider.getOrCreate(doc, {
+                        metadata: controller.connection,
+                        controller: controller?.controller,
+                        resourceUri: doc.uri
+                    });
                 }
+            }
 
-                const checkCompatible = () =>
-                    this.pythonInstaller.isProductVersionCompatible(Product.ipykernel, '>=6.0.0', interpreter);
```
Comment (DavidKutu) [inline src/client/debugger/jupyter/debuggingManager.ts:435]: No need for caching or the python installer anymore, this is just faster.

**Review by rchiodo — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/debugger/jupyter/debuggingManager.ts [hunk 2]
```diff
@@ -421,31 +411,35 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
...
-                });
-                this.cache.set(interpreter, result);
-                return result;
+            if (kernel) {
+                const code = 'import ipykernel\nprint(ipykernel.__version__)';
+                const output = await kernel.executeHidden(code);
+
+                if (output[0].text) {
+                    const majorVersion = Number(output[0].text.toString().charAt(0));
+                    const result = majorVersion >= 6;
```
Comment (roblourens) [inline src/client/debugger/jupyter/debuggingManager.ts:436]: Will work until ipykernel gets to version 10

Comment (DavidKutu) [reply src/client/debugger/jupyter/debuggingManager.ts:436]: OMG, thanks

Comment (DavidKutu) [reply src/client/debugger/jupyter/debuggingManager.ts:436]: addressed in https://github.com/microsoft/vscode-jupyter/pull/7695

**Pull request merged**